request_test.go

```diff
@@ -328,256 +378,604 @@ func TestPingRequestDefaultValues(t *testing.T) {
-		Index(validIndex).
-		Key(key)
+	req := NewDeleteRequest(defaultSpace)
+	req.Index("valid")
 	assertBodyEqual(t, refBuf.Bytes(), req)
 }
 
+func TestDeleteRequestSetters(t *testing.T) {
+	tests := []struct {
+		supportNames bool
+	}{
+		{false},
+		{true},
+	}
+
+	for _, test := range tests {
+		key := []interface{}{uint(923)}
+		var refBuf bytes.Buffer
+
+		resolver.nameUseSupported = test.supportNames
+
+		t.Run(fmt.Sprintf("NamesSupported_%v", test.supportNames), func(t *testing.T) {
+
+			refEnc := msgpack.NewEncoder(&refBuf)
+			err := RefImplDeleteBody(refEnc, &resolver, validSpace, validIndex, key)
+			if err != nil {
+				t.Errorf("An unexpected RefImplDeleteBody() error: %q", err.Error())
+				return
+			}
+
+			req := NewDeleteRequest(validSpace).
+				Index(validIndex).
+				Key(key)
+			assertBodyEqual(t, refBuf.Bytes(), req)
+		})
+	}
+}
+
 func TestUpdateRequestDefaultValues(t *testing.T) {
+	tests := []struct {
+		supportNames bool
+	}{
+		{false},
+		{true},
+	}
+
+	for _, test := range tests {
+		var refBuf bytes.Buffer
+
+		resolver.nameUseSupported = test.supportNames
+
+		t.Run(fmt.Sprintf("NamesSupported_%v", test.supportNames), func(t *testing.T) {
+
+			refEnc := msgpack.NewEncoder(&refBuf)
+			err := RefImplUpdateBody(refEnc, &resolver, validSpace, defaultIndex,
+				[]interface{}{}, []Op{})
+			if err != nil {
+				t.Errorf("An unexpected RefImplUpdateBody() error: %q", err.Error())
+				return
+			}
+
+			req := NewUpdateRequest(validSpace)
+			assertBodyEqual(t, refBuf.Bytes(), req)
+		})
+	}
+}
+
+func TestUpdateRequestSpaceByName(t *testing.T) {
 	var refBuf bytes.Buffer
 
+	resolver.nameUseSupported = true
+
 	refEnc := msgpack.NewEncoder(&refBuf)
-	err := RefImplUpdateBody(refEnc, validSpace, defaultIndex, []interface{}{}, []Op{})
+	err := RefImplUpdateBody(refEnc, &resolver, "valid", defaultIndex,
+		[]interface{}{}, []Op{})
 	if err != nil {
 		t.Errorf("An unexpected RefImplUpdateBody() error: %q", err.Error())
 		return
 	}
 
-	req := NewUpdateRequest(validSpace)
+	req := NewUpdateRequest("valid")
 	assertBodyEqual(t, refBuf.Bytes(), req)
 }
 
-func TestUpdateRequestSetters(t *testing.T) {
-	key := []interface{}{uint(44)}
-	refOps, reqOps := getTestOps()
+func TestUpdateRequestIndexByName(t *testing.T) {
 	var refBuf bytes.Buffer
 
+	resolver.nameUseSupported = true
+
 	refEnc := msgpack.NewEncoder(&refBuf)
-	err := RefImplUpdateBody(refEnc, validSpace, validIndex, key, refOps)
+	err := RefImplUpdateBody(refEnc, &resolver, defaultSpace, "valid",
+		[]interface{}{}, []Op{})
 	if err != nil {
 		t.Errorf("An unexpected RefImplUpdateBody() error: %q", err.Error())
 		return
 	}
 
-	req := NewUpdateRequest(validSpace).
-		Index(validIndex).
-		Key(key).
-		Operations(reqOps)
+	req := NewUpdateRequest(defaultSpace)
+	req.Index("valid")
 	assertBodyEqual(t, refBuf.Bytes(), req)
 }
 
+func TestUpdateRequestSetters(t *testing.T) {
+	tests := []struct {
+		supportNames bool
+	}{
+		{false},
+		{true},
```

Comment:
What do you actually test here with `supportNames == true`?

Reply:
Updated. Wrote new tests and removed changes from the existing ones. New tests check if corresponding Resolvers are called upon receiving `space` and `index` as strings, depending on the NameSpaceAndIndex support.